csharp/extractor/Semmle.Util/FileUtils.cs

```diff
@@ -50,5 +52,23 @@ public static void TryDelete(string file)
                 // Ignore
             }
         }
+
+        /// <summary>
+        /// Finds the path for the executable <paramref name="exe"/> based on the
+        /// <code>PATH</code> environment variable, and in the case of Windows the
+        /// <code>PATHEXT</code> environment variable.
+        /// 
+        /// Returns <code>null</code> of no path can be found.
+        /// </summary>
+        public static string FindExecutableOnPath(string exe)
+        {
+            var isWindows = Win32.IsWindows();
+            var paths = Environment.GetEnvironmentVariable("PATH").Split(isWindows ? ';' : ':');
+            var exes = isWindows
+                ? Environment.GetEnvironmentVariable("PATHEXT").Split(';').Select(ext => exe + ext)
```

Comment:
On my Windows machine, `PATHEXT` = `.COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC;.PY`

I'm not convinced that we should be searching for all of these combinations (e.g. `dotnet.py`), as I'm pretty sure it's only `dotnet.exe`.